rust/feature-flags/src/flag_matching.rs

```diff
@@ -228,39 +438,42 @@ impl FeatureFlagMatcher {
             .to_string();
         let hash_val = u64::from_str_radix(&hex_str, 16).unwrap();
 
-        hash_val as f64 / LONG_SCALE as f64
+        Ok(hash_val as f64 / LONG_SCALE as f64)
     }
 
     /// This function takes a feature flag and returns the key of the variant that should be shown to the user.
-    pub fn get_matching_variant(&self, feature_flag: &FeatureFlag) -> Option<String> {
-        let hash = self.get_hash(feature_flag, "variant");
+    pub async fn get_matching_variant(
+        &self,
+        feature_flag: &FeatureFlag,
+    ) -> Result<Option<String>, FlagError> {
+        let hash = self.get_hash(feature_flag, "variant").await?;
         let mut total_percentage = 0.0;
 
         for variant in feature_flag.get_variants() {
             total_percentage += variant.rollout_percentage / 100.0;
             if hash < total_percentage {
-                return Some(variant.key.clone());
+                return Ok(Some(variant.key.clone()));
             }
         }
-        None
+        Ok(None)
     }
 
     /// This function takes a feature flag and returns the key of the variant that should be shown to the user.
-    pub async fn get_person_properties_from_cache_or_db(
+    pub async fn get_person_properties_from_in_memory_cache_or_db(
         &mut self,
-        team_id: i32,
+        team_id: TeamId,
         distinct_id: String,
     ) -> Result<HashMap<String, Value>, FlagError> {
         // TODO: Do we even need to cache here anymore?
         // Depends on how often we're calling this function
         // to match all flags for a single person
 
         // TODO which of these properties do we need to cache?
-        if let Some(cached_props) = self.cached_properties.clone() {
+        if let Some(cached_person_props) = self.cached_person_properties.clone() {
             // TODO: Maybe we don't want to copy around all user properties, this will by far be the largest chunk
```

Comment:
ah I guess this is what confused you around property caching - this in django is a simple in memory cache for the life of the object, because accessing the matching result would otherwise make more DB queries. I don't think that replicates here directly anymore.

Reply:
GOT it